File: internal/sem/testdata/langmatrix/objc/Ledger.m
#import <Foundation/Foundation.h>

@interface Ledger : NSObject
@property (nonatomic) NSInteger total;
- (NSInteger)add:(NSInteger)amount;
@end

@implementation Ledger

- (NSInteger)add:(NSInteger)amount {
    return self.total + amount;
}

@end

NSInteger LedgerDouble(NSInteger amount) {
    Ledger *ledger = [[Ledger alloc] init];
    return [ledger add:amount] * 2;
}

File: internal/sem/testdata/langmatrix/swift/Ledger.swift
import Foundation

struct Ledger {
    var total: Int = 0

    func add(_ amount: Int) -> Int {
        return total + amount
    }
}

func ledgerDouble(_ amount: Int) -> Int {
    let ledger = Ledger()
    return ledger.add(amount) * 2
}
